strapi/strapi
# feat: add history versions endpoint

### What does it do?

Adds a route that lists the available history versions for a given set of query params.

Things that don't work yet:
- field-level permissions are ignored for now
- pagination: we just return a 10-item page for now
- I can't open entries with i18n enabled in the admin, so I wasn't able to test the locale matching. I suspect it may be an issue with my document middleware, as I'm using an API not meant to be used directly, so I'm waiting to implement the solution from #19377 before digging more into it.

### Why is it needed?

To list history versions in the frontend.

### How to test it?

- Create history versions by creating and updating content a bunch of times
- Make an API request to `/content-manager/history-versions`
- If you don't authenticate it as an admin user via the JWT, you should get an error
- If you don't give it both `contentType` and `documentId` params, you should get an error
- If you don't have permissions to view the `contentType` you passed it, you should get an error
- Otherwise you should see an array of relevant history versions in the data key, and a meta object with pagination
- The createdBy author should be populated and sanitized (no password hash)
